### DISCUSSION TITLE
Ability to change favicon

### DISCUSSION THREAD

**Discussion by segaura** [linuxserver/Heimdall#674] [Ideas]
Hallo.
In issue #132 [it has been suggested](https://github.com/linuxserver/Heimdall/issues/132#issuecomment-535121798) to mount the favicon files using docker this way

```
volumes:
  - /absolute/host/path/my-icon-16x16.png:/var/www/localhost/heimdall/public/favicon-16x16.png
  - /absolute/host/path/my-icon-32x32.png:/var/www/localhost/heimdall/public/favicon-32x32.png
  - /absolute/host/path/my-icon-96x96.png:/var/www/localhost/heimdall/public/favicon-96x96.png
  - /absolute/host/path/my-icon.ico:/var/www/localhost/heimdall/public/favicon.ico
```

If I do this I get these messages in the log

```
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-config: executing...
[cont-init.d] 20-config: exited 0.
[cont-init.d] 30-keygen: executing...
using keys found in /config/keys
[cont-init.d] 30-keygen: exited 0.
[cont-init.d] 50-config: executing...
New container detected, installing Heimdall
tar: public/favicon-16x16.png: Cannot open: File exists
tar: public/favicon-32x32.png: Cannot open: File exists
tar: public/favicon-96x96.png: Cannot open: File exists
tar: public/favicon.ico: Cannot open: File exists
tar: Exiting with failure status due to previous errors
Setting permissions
[cont-init.d] 50-config: exited 0.
[cont-init.d] 99-custom-files: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
```

and apparently this lead to a permanent and complete malfunctioning with this log

```
In PackageManifest.php line 168:

  The /var/www/localhost/heimdall/bootstrap/cache directory must be present and writable.
```

For those who managed to change favicon though docker, did you do something different or something else?

**Comment by RomainL972** [comment]
Hi,
We achieved this by putting the logos in the heimdall volume, for example in /config/logos, with all the files named favicon-<size>.png
Then, we created a script in /config/custom-cont-init.d/copy-logos.sh with the following content :
```bash
#!/bin/bash

cp /config/logos/* /var/www/localhost/heimdall/public/
```

**Reply by thellamapaul** [reply]
FYI with the nginx change favicons are now located in /app/www/public/

**Comment by esbeeb** [comment]
I made a custom Heimdall favicon set for my organization, which I'd like to contribute.  They're just green dots, nothing too fancy (since Heimdall is a "Start page" of sorts... "green means go" felt symbolic of that, and is easy for people to relate to, what with traffic lights having green meaning go).

These can simply be copied over the default files found in heimdall/public.  Harder to do in a Docker container.

[green_favicons_heimdall.zip](https://github.com/linuxserver/Heimdall/files/9721178/green_favicons_heimdall.zip)

**Comment by esbeeb** [comment]
Here's another custom favicon set, this time with the Wireguard logo. Great for when accessing Heimdall through a Wireguard tunnel (and your buttons/launchers are leading to different IPs/URLs than they are within your LAN).

[wireguard_favicons_heimdall.zip](https://github.com/linuxserver/Heimdall/files/13420784/wireguard_favicons_heimdall.zip)
